ethcore/src/verification/verification.rs

```diff
@@ -40,6 +40,35 @@ use types::{BlockNumber, header::Header};
 use types::transaction::SignedTransaction;
 use verification::queue::kind::blocks::Unverified;
 
+
+/// FIXME: @niklasad1 - remove this when and use `SystemTime::checked_add`
+/// when https://github.com/rust-lang/rust/issues/55940 is stabilized
+pub fn timestamp_checked_add(sys: SystemTime, d2: Duration) -> Result<SystemTime, BlockError> {
```

Comment:
Maybe let's rather check if `Duration` is within reasonable bounds instead? The logic below seems pretty complicated.